### PR TITLE
Static Marks improvements

### PR DESCRIPTION
**Pull request by rudeayelo**
## Description

Some additions to the Static Marks extension using convenient Raycast apis:

- Added the possibility to reference a local YAML file
- Show favicons for bookmarks
- Show categories (folders) for each bookmark as annotaions
- Show extended descriptions for bookmarks
- Categories and descriptions are taken into account in search results

## Screenshot

static-marks-1

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/static-marks-bookmarks/package.json [hunk 1]
```diff
@@ -22,27 +25,27 @@
...
     {
       "name": "static-mark-yaml-url",
-      "description": "The url where your Static Marks YAML file is located",
+      "description": "The absolute path or public url where your Static Marks YAML file is located",
       "type": "textfield",
-      "placeholder": "https://raw.githubusercontent.com/../bookmarks.yml",
-      "required": true,
-      "title": "Static Marks YAML URL"
+      "placeholder": "/Users/[user folder]/bookmarks.yml",
+      "required": false,
```
Comment (aerobless) [inline extensions/static-marks-bookmarks/package.json:31]: Why mark this as not required? Isn't it needed for the extension to work?

File: extensions/static-marks-bookmarks/src/index.tsx [hunk 2]
```diff
@@ -19,68 +26,151 @@ export default function Command() {
...
+  for (const [key, value] of Object.entries(json)) {
+    if (isValidBookmark(value)) {
+      const bookmarkUrls = getBookmarkUrl(value);
+      const bookmarkDescription = getBookmarkDescription(value);
+      const bookmarkParents = parents;
+      const bookmarkKeywords = [
+        ...parents.flatMap((parent) => parent.split(" ")),
+        ...(bookmarkDescription.length ? bookmarkDescription.split(" ") : []),
+      ];
+      console.log("bookmarkKeywords", bookmarkKeywords);
```
Comment (peduarte) [inline extensions/static-marks-bookmarks/src/index.tsx:131]: Left over `console.log`

File: extensions/static-marks-bookmarks/package.json (hunk 1 above)
Comment (peduarte) [reply extensions/static-marks-bookmarks/package.json:31]: Was gonna say this. Seems like it should be required: 
image

File: extensions/static-marks-bookmarks/src/index.tsx [hunk 3]
```diff
@@ -19,68 +26,151 @@ export default function Command() {
 }
 
 function SearchListItem({ searchResult }: { searchResult: LinkResult }) {
+  const parents = [{ text: searchResult.parents.join(" > "), icon: Icon.Folder }];
+
   return (
     <List.Item
       title={searchResult.name}
+      subtitle={searchResult.description}
+      icon={getFavicon(searchResult.url)}
```
Comment (peduarte) [inline extensions/static-marks-bookmarks/src/index.tsx:35]: I find that some square favicons can look a bit jarring. What do you think about applying a Mask?
```suggestion
      icon={getFavicon(searchResult.url, { mask: Image.Mask.Circle })}
```

Note: This is only supported in `@raycast/utils@1.4.11`

File: extensions/static-marks-bookmarks/package.json (hunk 1 above)
Comment (rudeayelo) [reply extensions/static-marks-bookmarks/package.json:31]: You're right, I played with the idea of making it default to the url placeholder, but finally went with the more explicit behaviour. Reverting that change right away.

Comment (aerobless) [reply extensions/static-marks-bookmarks/package.json:31]: Thanks for improving the extension btw. 😄 looks really cool!

File: extensions/static-marks-bookmarks/src/index.tsx (hunk 3 above)
Comment (rudeayelo) [reply extensions/static-marks-bookmarks/src/index.tsx:35]: Not so sure about this one, it adds consistency, but there are some beautiful favicons out there that look kind of odd after this treatment:

static-marks 2022-11-15 at 14 30 06

Comment (rudeayelo) [reply extensions/static-marks-bookmarks/src/index.tsx:35]: Wishing for an api that returns a favicon for a website with some padding and a white background....

Comment (rudeayelo) [reply extensions/static-marks-bookmarks/src/index.tsx:35]: With `Image.Mask.RoundedRectangle` it looks a bit better IMO, what do you think @peduarte?

static-marks 2022-11-15 at 15 45 12

Comment (rudeayelo) [reply extensions/static-marks-bookmarks/src/index.tsx]: I went with `Image.Mask.RoundedRectangle` and added a more convenient fallback 😄 

image

Comment (peduarte) [reply extensions/static-marks-bookmarks/src/index.tsx]: Looks good to me! Good call.

**Review by peduarte — APPROVED** [review]
Great improvements, looks good to me.

@aerobless As the original author, you happy with these changes or have any more feedback?

**Comment by aerobless** [comment]
Looks good to me @peduarte & @rudeayelo Thank you both for your work! 😄

**Review by aerobless — APPROVED** [review]

**Comment by rudeayelo** [comment]
Hey @peduarte, seems like some tests failed, did I miss something?

**Comment by peduarte** [comment]
@rudeayelo no, we're still working on that test internally.

**Pull request merged**